src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -214,6 +228,19 @@ private EnvVars getEnvironmentVars(Run<?, ?> build, TaskListener listener) {
         return environmentVars;
     }
 
+    static Secret getRestApiSecret(final String restApiKey) {
+        Secret secretKey = null;
+        List<UsernamePasswordCredentials> passwordCreds =
+                CredentialsProvider.lookupCredentials(UsernamePasswordCredentials.class, (Item)null, ACL.SYSTEM, Collections.<DomainRequirement>emptyList());
+        for (UsernamePasswordCredentials cred : passwordCreds) {
+            if (restApiKey.equals(cred.getUsername())) {
```

Comment:
We should also be looking at credential ID here (when we convert to secret text that would be the only option).